src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,289 @@
+        this._videoElement.currentTime = time;
+        log.debug("VTL: Thumbnail already loaded.", time);
+        return PPromise.resolve(time);
+      }
+    }
+
+    const manifest = this._player.getManifest();
+    if (manifest === null) {
+      return PPromise.reject(
+        new VideoThumbnailLoaderError("NO_MANIFEST",
+                                      "No manifest available."));
+    }
+    const contentInfos = getContentInfos(time, manifest);
+    if (contentInfos === null) {
+      return PPromise.reject(
+        new VideoThumbnailLoaderError("NO_TRACK",
+                                      "Couldn't find track for this time."));
+    }
+    const segment = contentInfos.representation.index.getSegments(time, 10)[0];
+    if (segment === undefined) {
+      return PPromise.reject(
+        new VideoThumbnailLoaderError("NO_THUMBNAIL",
+                                      "Couldn't find thumbnail."));
+    }
+
+    log.debug("VTL: Found thumbnail for time", time, segment);
+
+    if (this._currentJob !== undefined &&
+        this._currentJob.contentInfos.representation.id ===
+          contentInfos.representation.id &&
+        this._currentJob.contentInfos.adaptation.id ===
+          contentInfos.adaptation.id &&
+        this._currentJob.contentInfos.period.id ===
+          contentInfos.period.id &&
+        this._currentJob.contentInfos.manifest.id ===
+          contentInfos.manifest.id &&
+        this._currentJob.segment.id === segment.id) {
+      // The current job is already handling the loading for the wanted time
+      // (same thumbnail).
+      return this._currentJob.jobPromise;
+    }
+    this._currentJob?.stop();
+    return this.startJob(contentInfos, time, segment);
+  }
+
+  /**
+   * Dispose thumbnail loader.
+   * @returns {void}
+   */
+  dispose(): void {
+    this._currentJob?.stop();
+    this._loaders = {};
+    disposeMediaSource();
+  }
+
+  private startJob(contentInfos: IContentInfos,
+                   time: number,
+                   segment: ISegment
+  ): Promise<unknown> {
+    const loader = this._loaders[contentInfos.manifest.transport];
+    if (loader === undefined) {
+      const error =
+        new VideoThumbnailLoaderError("NO_LOADER",
+                                      "VideoThumbnailLoaderError: No " +
+                                      "imported loader for this transport type: " +
+                                      contentInfos.manifest.transport);
+      return PPromise.reject(error);
+    }
+    const killJob$ = new Subject();
+    const abortError$ = killJob$.pipe(
+      map(() => {
+        throw new VideoThumbnailLoaderError("ABORTED",
+                                            "VideoThumbnailLoaderError: Aborted job.");
+      })
+    );
+
+    const segmentLoader = createSegmentLoader(
+      loader.video.loader,
+      undefined,
+      { baseDelay: 0,
+        maxDelay: 0,
+        maxRetryOffline: 0,
+        maxRetryRegular: 0 }
```

Comment:
To see in the future if we need the fetchers at all, as we do not retry and only exploit the `data` event.
But it's nice that way also

Reply:
Yes, for sure.